opal/mca/accelerator/rocm/accelerator_rocm_module.c

```diff
@@ -0,0 +1,623 @@
+                            err, HIP_FUNCS.hipGetErrorString(err));
+        free(stream->stream);
+        free(stream);
+        return OPAL_ERROR;
+    }
+
+    return OPAL_SUCCESS;
+}
+
+static int mca_accelerator_rocm_destroy_stream(opal_accelerator_stream_t *stream)
+{
+
+    if (NULL == stream || NULL == stream->stream) {
+        return OPAL_ERR_BAD_PARAM;
+    }
+
+    hipError_t err = HIP_FUNCS.hipStreamDestroy(*(hipStream_t *)stream->stream);
+    if (hipSuccess != err) {
+        opal_output_verbose(10, opal_accelerator_base_framework.framework_output,
+                            "error while destroying the hipStream\n");
+        return OPAL_ERROR;
+    }
+    free(stream->stream);
+
+    return OPAL_SUCCESS;
+}
+
+static int mca_accelerator_rocm_synchronize_stream(opal_accelerator_stream_t *stream)
+{
+    if (NULL == stream || NULL == stream->stream) {
+        return OPAL_ERR_BAD_PARAM;
+    }
+
+    hipError_t err = HIP_FUNCS.hipStreamSynchronize(*(hipStream_t *)stream->stream);
+    if (hipSuccess != err) {
+        opal_output_verbose(10, opal_accelerator_base_framework.framework_output,
+                            "error while synchronizing hipStream\n");
+        return OPAL_ERROR;
+    }
+
+    return OPAL_SUCCESS;
+}
+
+static int mca_accelerator_rocm_create_event(opal_accelerator_event_t *event)
+{
+    if (NULL == event) {
+        return OPAL_ERR_BAD_PARAM;
+    }
+
+    event->event = malloc(sizeof(hipEvent_t));
+    if (NULL == event->event) {
+        free(event);
+        return OPAL_ERR_OUT_OF_RESOURCE;
+    }
+    hipError_t err = HIP_FUNCS.hipEventCreateWithFlags((hipEvent_t*)event->event,
+                                                       hipEventDisableTiming);
+    if (hipSuccess != err) {
+        opal_output_verbose(10, opal_accelerator_base_framework.framework_output,
+                            "error creating event\n");
+        free (event->event);
+        return OPAL_ERROR;
+    }
+
+    return OPAL_SUCCESS;
+}
+
+static int mca_accelerator_rocm_destroy_event(opal_accelerator_event_t *event)
+{
+    if (NULL == event || NULL == event->event){
+        return OPAL_ERR_BAD_PARAM;
+    }
+
+    hipError_t err = HIP_FUNCS.hipEventDestroy(*(hipEvent_t*)event->event);
+    if (hipSuccess != err) {
+        opal_output_verbose(10, opal_accelerator_base_framework.framework_output,
+                            "error destroying event\n");
+        return OPAL_ERROR;
+    }
+    free (event->event);
+
+    return OPAL_SUCCESS;
+}
+
+static int mca_accelerator_rocm_record_event(opal_accelerator_event_t *event,
+                                             opal_accelerator_stream_t *stream)
+{
+    if (NULL == event || NULL == event->event){
+        return OPAL_ERR_BAD_PARAM;
+    }
+    if (NULL == stream || NULL == stream->stream){
+        return OPAL_ERR_BAD_PARAM;
+    }
+
+    hipError_t err = HIP_FUNCS.hipEventRecord((hipEvent_t)event->event,
```

Comment:
I believe that the first parameter should be *(hipEvent_t *)event->event here.

Reply:
@edgargabriel is this still applicable?

Reply:
yes, I missed that one. Shall I make the patch or would you like to do that?